deeplearning4j-zoo/src/main/java/org/deeplearning4j/zoo/model/TinyYOLO.java

```diff
@@ -0,0 +1,287 @@
+                .gradientNormalization(GradientNormalization.RenormalizeL2PerLayer)
+                .gradientNormalizationThreshold(1.0)
+                .updater(new Adam.Builder().learningRate(1e-3).build())
+                .l2(0.00001)
+                .activation(Activation.IDENTITY)
+                .trainingWorkspaceMode(workspaceMode)
+                .inferenceWorkspaceMode(workspaceMode)
+                .graphBuilder()
+                .addInputs("input")
+                .setInputTypes(InputType.convolutional(inputShape[2], inputShape[1], inputShape[0]));
+
+        addLayers(graphBuilder, 1, 3, inputShape[0], 16, 2, 2);
+
+        addLayers(graphBuilder, 2, 3, 16, 32, 2, 2);
+
+        addLayers(graphBuilder, 3, 3, 32, 64, 2, 2);
+
+        addLayers(graphBuilder, 4, 3, 64, 128, 2, 2);
+
+        addLayers(graphBuilder, 5, 3, 128, 256, 2, 2);
+
+        addLayers(graphBuilder, 6, 3, 256, 512, 2, 1);
+
+        addLayers(graphBuilder, 7, 3, 512, 1024, 0, 0);
+        addLayers(graphBuilder, 8, 3, 1024, 1024, 0, 0);
+
+        int layerNumber = 9;
+        graphBuilder
+                .addLayer("convolution2d_" + layerNumber,
+                        new ConvolutionLayer.Builder(1,1)
+                                .nIn(1024)
+                                .nOut(nBoxes * (5 + numLabels))
+                                .weightInit(WeightInit.XAVIER)
+                                .stride(1,1)
+                                .convolutionMode(ConvolutionMode.Same)
+                                .weightInit(WeightInit.RELU)
+                                .activation(Activation.IDENTITY)
+                                .build(),
+                        "activation_" + (layerNumber - 1))
+                .addLayer("outputs",
+                        new Yolo2OutputLayer.Builder()
+                                .boundingBoxPriors(priors)
+                                .build(),
+                        "convolution2d_" + layerNumber)
+                .setOutputs("outputs").backprop(true).pretrain(false);
+
+        return graphBuilder.build();
+    }
+
+    @Override
+    public ComputationGraph init() {
+        ComputationGraph model = new ComputationGraph(conf());
+        model.init();
+
+        return model;
+    }
+
+    @Override
+    public ModelMetaData metaData() {
+        return new ModelMetaData(new int[][] {inputShape}, 1, ZooType.CNN);
+    }
+
+    @Override
+    public void setInputShape(int[][] inputShape) {
+        this.inputShape = inputShape[0];
+    }
+
+    /** Returns {@code inputShape[1] / 32}, where {@code inputShape[1]} should be a multiple of 32. */
+    public int getGridWidth() {
+        return inputShape[1] / 32;
+    }
+
+    /** Returns {@code inputShape[2] / 32}, where {@code inputShape[2]} should be a multiple of 32. */
+    public int getGridHeight() {
+        return inputShape[2] / 32;
+    }
+
+/*
+    // Import from a Keras model
+    @Override
```

Comment:
Why is this commented out? Perhaps we should just remove it completely?

Reply:
It's basically what I used to import the Keras model, might as well keep it for reference I thought, but if it's not useful, we can remove it, sure.

Reply:
Let's move it to documentation on either gh-pages or Basecamp.

Reply:
Perhaps the model-zoo page? https://deeplearning4j.org/model-zoo